app/controllers/WalletController.scala

```diff
@@ -21,6 +23,13 @@ class WalletController @Inject()(ws: WalletService, cc: ControllerComponents) ex
     }
   }
 
+  def getAllWithInfo(): Action[AnyContent] = Action.async { implicit request: Request[AnyContent] =>
+    val wallets: Seq[Wallet] = ws.loadAll
+    Future.sequence(wallets.map { w =>
+      es.requestUtxos(w.account.address).map(bxs => WalletInfo(w, bxs.map(_.value).sum))
+    }).map(xs => Ok(xs.asJson)).recover { case _ => InternalServerError }
+  }
+
```

Comment:
should this stuff be moved to WalletService ?

Reply:
Yes